modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -4479,45 +4518,88 @@ else if (!CU.affinityNode(ctx.discovery().localNode(), descCfg.getNodeFilter()))
+        Collection<IgniteInternalCache> caches = manageStatisticsCaches(cacheNames);
 
-        Collection<String> globalCaches = new ArrayList<>(caches.size());
+        Collection<String> globalCaches = new HashSet<>(caches.size());
 
-        if (!cacheNames().containsAll(caches))
-            throw new IgniteCheckedException("One or more cache descriptors not found [caches=" + caches + ']');
+        for (IgniteInternalCache cache : caches) {
+            cache.context().statisticsEnabled(enabled);
 
-        for (String cacheName : caches) {
-            IgniteInternalCache cache = cache(cacheName);
+            if(!cache.context().isLocal())
+                globalCaches.add(cache.name());
+        }
 
-            if (cache == null)
-                throw new IgniteCheckedException("Cache not found [cacheName=" + cacheName + ']');
+        if (globalCaches.isEmpty())
+            return;
 
-            cache.context().statisticsEnabled(enabled);
+        CacheStatisticsModeChangeMessage msg = new CacheStatisticsModeChangeMessage(UUID.randomUUID(), globalCaches, enabled);
+
+        EnableStatisticsFuture fut = new EnableStatisticsFuture(msg.requestId());
+
+        manageStatisticsFuts.put(msg.requestId(), fut);
+
+        ctx.grid().context().discovery().sendCustomEvent(msg);
+
+        fut.get();
+    }
 
-            if (!cache.context().isLocal())
-                globalCaches.add(cacheName);
+    /**
+     * Clear statistics globally for the caches
+     *
+     * @param cacheNames Collection of cache names.
+     */
+    public void clearStatistics(Collection<String> cacheNames) throws IgniteCheckedException {
+        Collection<IgniteInternalCache> caches = manageStatisticsCaches(cacheNames);
+
+        Collection<String> globalCaches = new HashSet<>(caches.size());
+
+        for (IgniteInternalCache cache : caches) {
+            if(!cache.context().isLocal() && cache.clusterMxBean().isStatisticsEnabled())
+                globalCaches.add(cache.name());
         }
 
         if (globalCaches.isEmpty())
             return;
 
-        UUID reqId = UUID.randomUUID();
+        CacheStatisticsClearMessage msg = new CacheStatisticsClearMessage(UUID.randomUUID(), globalCaches);
 
-        EnableStatisticsFuture fut = new EnableStatisticsFuture(reqId);
+        EnableStatisticsFuture fut = new EnableStatisticsFuture(msg.requestId());
 
-        enableStatisticsFuts.put(reqId, fut);
-
-        CacheStatisticsModeChangeMessage msg = new CacheStatisticsModeChangeMessage(reqId, globalCaches, enabled);
+        manageStatisticsFuts.put(msg.requestId(), fut);
 
         ctx.grid().context().discovery().sendCustomEvent(msg);
 
         fut.get();
     }
 
+    /**
+     *
+     */
+    private Collection<IgniteInternalCache> manageStatisticsCaches(Collection<String> caches)
+        throws IgniteCheckedException {
+        assert caches != null;
+
+        Collection<IgniteInternalCache> res = new ArrayList<>(caches.size());
```

Comment:
You can create `HashSet` instead of `ArrayList` and use it in further code instead of creating another `HashSet`.

Reply:
Eventually, I have to transform the collection of caches into a collection of strings that contains caches names and pass it to the message.

Reply:
Agree, my bad.